src/main/java/org/datadog/jenkins/plugins/datadog/publishers/DatadogQueuePublisher.java

```diff
@@ -70,23 +72,55 @@ protected void doRun() throws Exception {
             Map<String, Set<String>> tags = DatadogUtilities.getTagsFromGlobalTags();
             // Add JenkinsUrl Tag
             tags = TagsUtil.addTagToTags(tags, "jenkins_url", DatadogUtilities.getJenkinsUrl());
-
             long size = 0;
             long buildable = queue.countBuildableItems();
             long pending = queue.getPendingItems().size();
             long stuck = 0;
             long blocked = 0;
+            String hostname = DatadogUtilities.getHostname(null);
             final Queue.Item[] items = queue.getItems();
             for (Queue.Item item : items) {
+                Map<String, Set<String>> job_tags = DatadogUtilities.getTagsFromGlobalTags();
+                job_tags = TagsUtil.addTagToTags(job_tags, "jenkins_url", DatadogUtilities.getJenkinsUrl());
+                
+                String job_name;
+                Task task = item.task;
+                if (task instanceof FreeStyleProject){ 
+                    job_name = task.getFullDisplayName();
+                    TagsUtil.addTagToTags(job_tags, "job_name", job_name);
```

Comment:
Let's add an else clause that adds `job_name:unknown`. This will make queries better on Datadog

Reply:
Good idea, updated!